### ISSUE TITLE
Gpu inference

### ISSUE THREAD

**Issue by puppetm4st3r** [diffusion-studio/vits-web#3]
is there a way to use webgpu or multithread cpu inference? or that depends on the browser/os host? regards

**Comment by k9p5** [comment]
WebGPU browser compatibility is still rather limited, however it should most definitely be possible. I'll be working on it asap.

**Comment by puppetm4st3r** [comment]
thanks, and for cpu inference? it uses multithread inference? when i try raw onxx on wasm cannot do it with multithread inference,  no errors just did not do anything

**Comment by k9p5** [comment]
Yes it can be multithreaded but you need to enable cross-origin isolation.

**Comment by C-Loftus** [comment]
This is a great idea and would be very useful.

I would like to look into this or help you out. Have there been any updates on this or pointers for where to start? I wasn't sure how much of this needs to be done in WebGPU via JS or inside the WebAssembly. I also wasn't sure whether or not this will touch the build process in the [piper wasm](https://github.com/diffusionstudio/piper-wasm) repo.

**Comment by raymondtri** [comment]
I'm also interested in helping with this where possible.

From my understanding the blocker is related to the actual piper library and not vits. ONNX runtime has gpu support of course, and piper can be ran in GPU locally, but it looks like there's no compiled WASM bundle of piper that is built for gpu support which would run on the ONNX runtime.

I'm pretty new to WASM especially GPU WASM, so I could be wrong.

**Comment by C-Loftus** [comment]
> From my understanding the blocker is related to the actual piper library and not vits. ONNX runtime has gpu support of course, and piper can be ran in GPU locally, but it looks like there's no compiled WASM bundle of piper that is built for gpu support which would run on the ONNX runtime.

I think that is correct. As I have look into this further, it also seems based on https://github.com/ken107/read-aloud/issues/424#issuecomment-2463426855 that even in piper itself within a standard desktop runtime there are issues with GPU utilization. So I think this might be blocked on piper before pursuing web / wasm issues further.

**Comment by raymondtri** [comment]
https://github.com/rhasspy/piper/issues/598
